metafacture-biblio/src/main/java/org/metafacture/biblio/marc21/Marc21Encoder.java

```diff
@@ -190,7 +196,22 @@ public void literal(final String name, final String value) {
         }
     }
 
-    private void processLiteralInLeader(final String name, final String value) {
+    private void processLeaderAsOneLiteral(final String value) {
+        if (value.length() != Iso2709Constants.RECORD_LABEL_LENGTH) {
+            throw new FormatException(
+                    "leader literal must contain " + Iso2709Constants.RECORD_LABEL_LENGTH + "  characters:" + value);
+        }
+        processLeaderAsSubfields(Marc21EventNames.RECORD_STATUS_LITERAL, String.valueOf(value.charAt(Iso2709Constants.RECORD_STATUS_POS)));
+        processLeaderAsSubfields(Marc21EventNames.RECORD_TYPE_LITERAL, String.valueOf(value.charAt(Iso2709Constants.IMPL_CODES_START)));
+        processLeaderAsSubfields(Marc21EventNames.BIBLIOGRAPHIC_LEVEL_LITERAL, String.valueOf(value.charAt(Iso2709Constants.IMPL_CODES_START + 1)));
+        processLeaderAsSubfields(Marc21EventNames.TYPE_OF_CONTROL_LITERAL, String.valueOf(value.charAt(Iso2709Constants.IMPL_CODES_START + 2)));
+        processLeaderAsSubfields(Marc21EventNames.CHARACTER_CODING_LITERAL, String.valueOf(value.charAt(Iso2709Constants.RECORD_STATUS_POS + Iso2709Constants.IMPL_CODES_LENGTH)));
+        processLeaderAsSubfields(Marc21EventNames.ENCODING_LEVEL_LITERAL, String.valueOf(value.charAt(Iso2709Constants.SYSTEM_CHARS_START)));
+        processLeaderAsSubfields(Marc21EventNames.CATALOGING_FORM_LITERAL, String.valueOf(value.charAt(Iso2709Constants.SYSTEM_CHARS_START + 1)));
+        processLeaderAsSubfields(Marc21EventNames.MULTIPART_LEVEL_LITERAL, String.valueOf(value.charAt(Iso2709Constants.SYSTEM_CHARS_START + 2)));
+    }
+
+    private void processLeaderAsSubfields(final String name, final String value) {
```

Comment:
Maybe add `processLeaderAsSubfields(String, char)` to avoid `char` -> `String` -> `char` conversion?

Reply:
good idea - did so.